lib/bb/client.rb

```diff
@@ -51,5 +53,19 @@ def get_download_report(doctype, header_callback, yielder)
       uri = URI.join(config.base_path, "bluebutton/bbreport/#{doctype}")
       streaming_get(uri, token_headers, header_callback, yielder)
     end
+
+    private
+
+    def refresh_final?(attrs)
+      attrs.all? { |e| e[:status].present? }
+    end
+
+    def log_refresh_errors(attrs)
+      failed = attrs.select { |e| e[:status] == 'ERROR' }.map { |e| e[:extract_type] }
+      if failed.present?
+        log_message_to_sentry('Final health record refresh contained one or more error statuses', :warn,
+                              refresh_failures: failed.sort)
```

Comment:
Everything looks good, but can this be graphed in graphana as effectively as maybe iterating through the list of failures and logging each individually?